pyccel/ast/basic.py

```diff
@@ -180,24 +180,7 @@ def is_attribute_of(self, search_node, excluded_nodes = ()):
         -------
         bool
         """
-        if self._recursion_in_progress or len(self._user_nodes) == 0:
-            return []
-        else:
-            self._recursion_in_progress = True
-
-            if any(p is search_node for p in self._user_nodes):
-                self._recursion_in_progress = False
-                return True
-
-            for p in self._user_nodes:
-                if not self.ignore(p) and not isinstance(p, excluded_nodes):
-                    res = p.is_attribute_of(search_node, excluded_nodes = excluded_nodes)
-                    if res:
-                        self._recursion_in_progress = False
-                        return True
-
-            self._recursion_in_progress = False
-            return False
+        return node.is_user_of(self)
```

Comment:
The argument of this function should be called `node`, or this variable should be changed to `search_node`. I prefer the former option.

Reply:
Thanks. The docstrings now need updating...

Reply:
Fixed 6d78abaa